app/src/main/java/de/danoeh/antennapod/dialog/IntraFeedSortDialog.java

```diff
@@ -16,35 +16,63 @@
     @NonNull
     protected Context context;
 
+    private String[] localSortItems;
+    private SortOrder[] localSortValues;
+
+    private String[] commonSortItems;
+    private SortOrder[] commonSortValues;
+
     public IntraFeedSortDialog(@NonNull Context context, @Nullable SortOrder sortOrder) {
         this.context = context;
         this.currentSortOrder = sortOrder;
+        setSortOptions();
     }
 
-    public void openDialog() {
-        final String[] items = context.getResources().getStringArray(R.array.feed_episodes_sort_options);
-        final String[] valueStrs = context.getResources().getStringArray(R.array.feed_episodes_sort_values);
-        final SortOrder[] values = new SortOrder[valueStrs.length];
-        for (int i = 0; i < valueStrs.length; i++) {
-            values[i] = SortOrder.valueOf(valueStrs[i]);
-        }
+    private void setSortOptions() {
```

Comment:
Please move the initialization directly to the constructor. Then you can also make the values `final`.